Rakefile

```diff
@@ -71,6 +75,63 @@ def re_run_basic_smoke_test
   sh beaker
 end
 
+def get_passing_json(current_branch_trunk)
+  jenkins_url = "https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/#{current_branch_trunk}/view/Suite/job/platform_puppet-agent_intn-van-promote_suite-daily-promotion-#{current_branch_trunk}/lastSuccessfulBuild/api/json"
+  @memoized_passing_json ||= Net::HTTP.get(URI.parse(jenkins_url))
+end
+def get_passing_puppetagent_sha(current_branch_trunk)
+  get_passing_json(current_branch_trunk)
+  JSON.parse(@memoized_passing_json)['actions'][0]['parameters'].find{|x| x['name'] == 'SUITE_COMMIT'}['value']
```

Comment:
Calling a method to mutate a variable and then directly operating on that variable is something that I would probably avoid even in an OO language. `get_passing_json` returns the output of `@memoized_passing_json` so I would either save the return of `get_passing_json` as a local value, or call it directly in the `JSON.parse` parameter list. That way `get_passing_*_sha` methods aren't coupled to the internal structure of how `get_passing_sha` is memoizing the data.